codegen/src/generator/mod.rs

```diff
@@ -262,11 +287,8 @@ fn generate_struct_fields<P>(service: &Service, shape: &Shape, shape_name: &str,
 
 impl Operation {
     pub fn error_type_name(&self) -> String {
-        match &self.name[..] {
-            // EC2 has a different struct type called CancelSpotFleetRequestsError
-            "CancelSpotFleetRequests" => "CancelSpotFleetRequestsErrorType".to_owned(),
-            _ => format!("{}Error", self.name)
-        }
+        let type_name = mutate_type_name(&self.name);
+        format!("{}Error", type_name)
```

Comment:
If I understand correctly, one of the type names S3 provides is `Error`, which is turned into `S3Error` via `mutate_type_name`. Does this function return `S3ErrorError` then? Or is `S3Error `never used with this function?